ale_linters/cpp/gcc.vim

```diff
@@ -9,13 +9,15 @@ let g:loaded_ale_linters_cpp_gcc = 1
 
 " Set this option to change the GCC options for warnings for C.
 if !exists('g:ale_cpp_gcc_options')
-    let g:ale_cpp_gcc_options = '-Wall'
+  " added c++14 standard support
+  " POSIX thread and standard c++ thread and atomic library Linker
+    let g:ale_cpp_gcc_options = '-std=c++14 -ldl -lpthread -lm -latomic -Wall -Wextra -Wshadow -pedantic'
```

Comment:
Another thing...! for c++14, In operating system, gcc aka g++ version must be 4.9.0 or greater.

Reference : https://isocpp.org/blog/2014/04/gcc-4.9.0